.github/workflows/test.yml

```diff
@@ -1,25 +1,38 @@
 ---
-name: UnitTesting
-
-on: [push, pull_request]
-
+name: Testing
+
+env:
+  DOCKER_HUB_ID: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+  DOCKER_HUB_PASSWORD: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+  TAG: "0.0.0" # needed for goreleaser windows builds
+  GO_VERSION: '1.14.4'
+
+on:
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - README.md
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  unit-test-linux:
+    name: Linux unit tests
+    runs-on: ubuntu-20.04
 
     steps:
       - uses: actions/checkout@v2
 
       - name: Install Go
```

Comment:
If we are using containers then this step isn't needed. It means we need to add a target called `ci/gcov2lcov` and then have a step that in a container runs the install `gcov2lcov` and runs the `gcov2lcov -infile=coverage.out -outfile=lcov.info`. We should also pin `gcov2lcov` with go modules.